src/link_prober/LinkProber.cpp

```diff
@@ -439,6 +439,18 @@ void LinkProber::handleTimeout(boost::system::error_code errorCode)
     if (mTxSeqNo != mRxSelfSeqNo && mTxSeqNo != mRxPeerSeqNo) {
         // post unknown event
         mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpUnknownEvent());
+        mIcmpUnknownEventCount++;
+    } 
+
+    mIcmpPacketCount++;
+    if (mIcmpPacketCount % mMuxPortConfig.getNegativeStateChangeRetryCount() == 0) {
+        boost::asio::io_service::strand &strand = mLinkProberStateMachine.getStrand();
+        boost::asio::io_service &ioService = strand.context();
+        ioService.post(strand.wrap(boost::bind(
+            &LinkProberStateMachine::handlePckLossRatioUpdate,
+            &mLinkProberStateMachine,
+            (static_cast<double_t>(mIcmpUnknownEventCount) / mIcmpPacketCount)
```

Comment:
Please protect against divide by 0

Reply:
Updated the code to post counts instead of ratio.